### PR TITLE
Twitter scrapper

### LINKED ISSUE #752: Script to scrap Twitter Data
# Description

Add a Script to scrap Twitter Data in a neat-CSV file with the following fields: 

1. Tweets 
2. Retweets
3. Likes
4. Pinned (if present)
5. Hashtags, Videos, Photos

## Language

- [X] Go
- [X] Javascript
- [X] Python

## Checklist:

| name            | about                                                  | labels  | assignees                                                      |
| --------------- | ------------------------------------------------------ | ------- | -------------------------------------------------------------- |
| Twitter Scrapper | Scraps everything off your Twitter timeline | MWOC, Python, Javascript, Golang, enhancement | |

### PR DESCRIPTION
**Pull request by DiegoCaraballo**
# Description

This script extracts all Tweets, Retweets, Images, Videos, Hashtags,
Likes and Pinned tweet of a specific user in a .csv file.

Fixes #752

Replace `issue_no` with the issue number which is fixed in this PR

## Type of change

_Please delete options that are not relevant._

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Documentation Update

# Checklist:

- [x] My code follows the style guidelines(Clean Code) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have created a helpful and easy to understand `README.md`
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests/screenshots(if any) that prove my fix is effective or that my feature works

### CODE REVIEW

File: Python/Twitter_Scrapper/twitter_scrapper.py [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+
+import tweepy
+from bs4 import BeautifulSoup
+from selenium import webdriver
+
+# Authenticate to Twitter
+CONSUMER_KEY = "<your-consumer-or-API-key-goes-here>"
+CONSUMER_SECRET = "<your-consumer-or-API-secret-goes-here>"
+ACCESS_KEY = "<your-access-key-goes-here>"
+ACESS_SECRET = "<your-access-secret-goes-here>"
```
Comment (HarshCasper) [inline Python/Twitter_Scrapper/twitter_scrapper.py:19]: Can you please add all of this in a `.env` file and use a module like `decouple` to import them in the Code? Also please don't push the `.env` into the PR and instead put a `.env.example` just for the other Devs to understand on what values do they need to fill.

File: Python/Twitter_Scrapper/twitter_scrapper.py [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+auth = tweepy.OAuthHandler(CONSUMER_KEY, CONSUMER_SECRET)
+auth.set_access_token(ACCESS_KEY, ACESS_SECRET)
+api = tweepy.API(auth)
+
+URL = "https://twitter.com"
+PATH_CHROME_DRIVER = "<===============ENTER YOUR CHROME DRIVER PATH===========>"
```
Comment (HarshCasper) [inline Python/Twitter_Scrapper/twitter_scrapper.py:25]: Add this into the `.env` as well.

File: Python/Twitter_Scrapper/README.md [hunk 3]
```diff
@@ -0,0 +1,26 @@
+## Twitter Scrapper
+
+This script extracts all Tweets, Retweets, Images, Videos, Hashtags, Likes and Pinned tweet of a specific user in a .csv file.
+
+![How to use the script](https://i.ibb.co/mcjK65f/Tweeter-Scrapper.png)
```
Comment (HarshCasper) [inline Python/Twitter_Scrapper/README.md:5]: Can you please add the snapshot on the bottom side of the documentation ?

File: Python/Twitter_Scrapper/README.md [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+4. Get the credentials from the app created by you on twitter developer account and enter them here.
+5. Also add the path to your chrome driver in the PATH_CHROME_DRIVER constant
+6.  Hurray.! You're ready to use the script to extracts all Tweets, Retweets, Images, Videos, Hashtags and Likes of a specific user.
+
+### Running a File
+
+1.  Run the Script  `python twitter_scrapper.py`
+3.  Enter the username you want the information from.
+4. Finally, you need to enter a Twitter username and password in order to extract the pinned Tweet (if it exists)
+5. The .csv file is downloaded to the folder where the file runs.
```
Comment (HarshCasper) [inline Python/Twitter_Scrapper/README.md:26]: Keep `.csv` in backticks.

File: Python/Twitter_Scrapper/README.md [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+3. Open the script in any text editor/IDE
+4. Get the credentials from the app created by you on twitter developer account and enter them here.
+5. Also add the path to your chrome driver in the PATH_CHROME_DRIVER constant
+6.  Hurray.! You're ready to use the script to extracts all Tweets, Retweets, Images, Videos, Hashtags and Likes of a specific user.
+
+### Running a File
+
+1.  Run the Script  `python twitter_scrapper.py`
+3.  Enter the username you want the information from.
+4. Finally, you need to enter a Twitter username and password in order to extract the pinned Tweet (if it exists)
```
Comment (HarshCasper) [inline Python/Twitter_Scrapper/README.md:25]: Can we add a function to handle the exception if the User doesn't exist?

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Great Work @dcorriveau ! 

Few changes required. Rest are good to merge.

**Comment by dcorriveau** [comment]
> Great Work @dcorriveau !
> 
> Few changes required. Rest are good to merge.

I think you might have meant @DiegoCaraballo not myself... :)

File: Python/Twitter_Scrapper/README.md (hunk 4 above)
Comment (DiegoCaraballo) [reply Python/Twitter_Scrapper/README.md:26]: Hello @HarshCasper 
What do you mean by that?

File: Python/Twitter_Scrapper/README.md (hunk 3 above)
Comment (DiegoCaraballo) [reply Python/Twitter_Scrapper/README.md]: Add the image on the bottom side

File: Python/Twitter_Scrapper/twitter_scrapper.py (hunk 2 above)
Comment (DiegoCaraballo) [reply Python/Twitter_Scrapper/twitter_scrapper.py]: Add the file .env.example

File: Python/Twitter_Scrapper/twitter_scrapper.py (hunk 1 above)
Comment (DiegoCaraballo) [reply Python/Twitter_Scrapper/twitter_scrapper.py]: Add the file .env.example

**Comment by HarshCasper** [comment]
> > Great Work @dcorriveau !
> > Few changes required. Rest are good to merge.
> 
> I think you might have meant @DiegoCaraballo not myself... :)

Sincere apologies 😄

**Review by HarshCasper — APPROVED** [review]

**Comment by HarshCasper** [comment]
Thanks @DiegoCaraballo 

Merged 😄

**Comment by HarshCasper** [comment]
@all-contributors please add @DiegoCaraballo for code

**Comment by HarshCasper** [comment]
@all-contributors please add @DiegoCaraballo for documentation

**Pull request merged**